### PR TITLE
Add nodeport to ports configuration

### PR DESCRIPTION
**Pull request by kostya9**
Add nodePort entry to ports configuration.

This allows us to specify the exact port to use by the Service (type=NodePort).

### CODE REVIEW

**Comment by TylerHelmuth** [comment]
I am out for Memorial Day but will review tomorrow.

File: charts/opentelemetry-collector/values.schema.json [hunk 1]
```diff
@@ -215,6 +215,9 @@
             "hostPort": {
               "type": "integer"
             },
+            "nodePort": {
+                "type": "integer"
+              },
```
Comment (TylerHelmuth) [inline charts/opentelemetry-collector/values.schema.json:220]:
```suggestion
            },
```

**Review by TylerHelmuth — COMMENTED** [review]
Can you add in the values.yaml in the `port` section a commented out `nodePort:` for each port?  Also a comment that nodePort is only used with deployment mode.

File: charts/opentelemetry-collector/Chart.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v2
 name: opentelemetry-collector
-version: 0.18.0
+version: 0.19.0
```
Comment (TylerHelmuth) [inline charts/opentelemetry-collector/Chart.yaml:3]:
```suggestion
version: 0.18.1
```

**Comment by kostya9** [comment]
Thanks for the review, addressed your comments in the last commit

File: charts/opentelemetry-collector/values.yaml [hunk 3]
```diff
@@ -155,36 +155,43 @@ ports:
     containerPort: 4317
     servicePort: 4317
     hostPort: 4317
+    # nodePort could only be used with the deployment mode
```
Comment (TylerHelmuth) [inline charts/opentelemetry-collector/values.yaml:158]:
```suggestion
    # nodePort can only be used with deployment mode
```

**Review by TylerHelmuth — APPROVED** [review]

File: charts/opentelemetry-collector/values.yaml (hunk 3 above)
Comment (kostya9) [reply charts/opentelemetry-collector/values.yaml]: fixed

**Comment by TylerHelmuth** [comment]
@dmitryax please take a look when you're free.

**Comment by TylerHelmuth** [comment]
@kostya9 can you take a look at the Dmitrii's feedback and also merge latest from main?  I'll need to bump the chart version and regenerate the examples probably.

**Pull request closed without merging**